pkg/capabilities/v2/capability.go

```diff
@@ -0,0 +1,164 @@
+package v2
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/invopop/jsonschema"
+	jsonvalidate "github.com/santhosh-tekuri/jsonschema/v5"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+)
+
+type CapabilityResponse[O any] struct {
+	Value O
+	Err   error
+}
+
+type CapabilityRequest[I, C any] struct {
```

Comment:
How do those work in a remote setting? Does the receiver always know what type to expect or is there some type info encoded too?

Reply:
I'd imagine it's the same, capabilities should have a input/config/output schema they abide to.

Reply:
@bolekk This is purely a client-side feature (i.e. inside the capability), anything accessing the capability (either via the Remote API or via the workflow engine would continue to do so via the lower-level, values-based API.